### PR TITLE
Add snapshotting/commitlog upgrade instructions to CHANGELOG

### PR DESCRIPTION
**Pull request by richardartoul**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+
+The redesign is **backwards compatible** but not **forwards compatible**. This means that you should be able upgrade your < 0.7.0 clusters to 0.7.0 with no issues, but you will not be able to downgrade without taking some additional steps. Note that the first bootstrap after the upgrade may take longer than usual, but subsequent bootstraps should be just as fast as they used to be, or even faster.
+
+### Troubleshotting and Rolling Back
+
+If you run into any issues with the upgrade or need to downgrade to a previous version for any reason, follow these steps:
+
+1. Stop the node that is having trouble with the upgrade or that you're trying to downgrade.
+2. Modify the `bootstrappers` config in the M3DB YAML file from `filesystem, commitlog, peers, uninitialize_topology` to `filesystem, peers, commitlog, uninitialized_topology`. This will force the node to bootstrap from its peers instead of the local snapshot and commitlog files it has on disk, bypassing any issues related to file incompatibility between versions.
```
Comment (schallert) [inline CHANGELOG.md]: `s/uninitialize_topology/uninitialized_topology/`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+3. Turn the node back on and wait for it to finish bootstrapping and snapshotting. Once everything looks stable, change the config back to `filesystem, commitlog, peers, uninitialize_topology` so  that the next time the node is restarted it will default to using the snapshot and commitlog files.
```
Comment (schallert) [inline CHANGELOG.md]: Same here?

**Comment by schallert** [comment]
Would it be possible to include links to some of the PRs like in releases below or were they too spread out?

**Comment by richardartoul** [comment]
@schallert There were a bunch of P.Rs but I linked to the main one

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
 
+# 0.7.0 (UNRELEASED)
+
+## Migration Disclaimer
+
+Version 0.7.0 of M3 includes a redesign of the snapshotting and commitlog components (#1384 among others). This redesign was critical to improve M3DB's consistency guarantees, reduce the amount of disk space that is wasted by commitlogs, and enable future feature development to support writing data at arbitrary times.
+
+The redesign is **backwards compatible** but not **forwards compatible**. This means that you should be able upgrade your < 0.7.0 clusters to 0.7.0 with no issues, but you will not be able to downgrade without taking some additional steps. Note that the first bootstrap after the upgrade may take longer than usual, but subsequent bootstraps should be just as fast as they used to be, or even faster.
+
+### Troubleshotting and Rolling Back
```
Comment (arnikola) [inline CHANGELOG.md]: nit: Troubleshooting*

**Pull request merged**